install/outline-install.sh

```diff
@@ -0,0 +1,154 @@
+wget -qO- https://packages.redis.io/gpg | gpg --dearmor -o /etc/apt/keyrings/redis-archive-keyring.gpg
+echo "deb [signed-by=/etc/apt/keyrings/redis-archive-keyring.gpg] https://packages.redis.io/deb $(lsb_release -cs) main" >/etc/apt/sources.list.d/redis.list
+msg_ok "Set up Redis Repository"
+
+msg_info "Installing Node.js/Yarn"
+$STD apt-get update
+$STD apt-get install -y nodejs
+$STD npm install -g yarn
+msg_ok "Installed Node.js/Yarn"
+
+msg_info "Installing Redis"
+$STD apt-get install -y redis
+sed -i 's/^bind .*/bind 0.0.0.0/' /etc/redis/redis.conf
+systemctl enable -q --now redis-server.service
+msg_ok "Installed Redis"
+
+msg_info "Setting up PostgreSQL DB"
+SECRET_KEY="$(openssl rand -hex 32)"
+UTILS_SECRET="$(openssl rand -hex 32)"
+DB_NAME=outlinedb
+DB_USER=outline
+DB_PASS="$(openssl rand -base64 18 | tr -d '/' | cut -c1-13)"
+DATABASE_URL="postgres://${DB_USER}:${DB_PASS}@localhost:5432/${DB_NAME}"
+$STD sudo -u postgres psql -c "CREATE ROLE $DB_USER WITH LOGIN PASSWORD '$DB_PASS';"
+$STD sudo -u postgres psql -c "ALTER ROLE $DB_USER WITH CREATEDB;"
+$STD sudo -u postgres psql -c "ALTER ROLE $DB_USER SET timezone TO 'UTC';"
+
+cat <<EOF >~/outline.creds
+Outline-Credentials
+Outline Database User: $DB_USER
+Outline Database Password: $DB_PASS
+Outline Database Name: $DB_NAME
+Outline Secret: $SECRET_KEY
+Outline Utils Secret: $UTILS_SECRET
+EOF
+msg_ok "Set up PostgreSQL DB"
+
+
+read -r -p "Would you like to add Adminer? <y/N> " adminer_prompt
+if [[ "${adminer_prompt,,}" =~ ^(y|yes)$ ]]; then
+  msg_info "Installing Adminer"
+  $STD apt install -y adminer
+  $STD a2enconf adminer
+  systemctl reload apache2
+  IP=$(hostname -I | awk '{print $1}')
+cat <<EOF >>~/outline.creds
+
+Adminer Interface: $IP/adminer/
+Adminer System: PostgreSQL
+Adminer Server: localhost:5432
+Adminer Username: $DB_USER
+Adminer Password: $DB_PASS
+Adminer Database: $DB_NAME
+EOF
+  msg_ok "Installed Adminer"
+fi
+
+read -r -p "Enter the public url for your Outline instance (e.g., https://outline.your-domain.tld)): " URL
+msg_info "Installing Outline (Patience)"
+cd /opt
+RELEASE_TAG=$(curl -s https://api.github.com/repos/outline/outline/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }')
+RELEASE=${RELEASE_TAG#v}
+wget -q "https://github.com/outline/outline/archive/refs/tags/${RELEASE_TAG}.zip"
+unzip -q ${RELEASE_TAG}.zip
+mv outline-${RELEASE} /opt/outline
+cd /opt/outline
+
+$STD yarn install --no-optional --frozen-lockfile
+$STD yarn cache clean
+$STD yarn build
+
+rm -rf ./node_modules
+$STD yarn install --production=true --frozen-lockfile
+$STD yarn cache clean
+
+FILE_STORAGE_LOCAL_ROOT_DIR=/var/lib/outline/data
+mkdir -p $FILE_STORAGE_LOCAL_ROOT_DIR
+
+cat <<EOF >/opt/outline/.env
+NODE_ENV=production
+SECRET_KEY=$SECRET_KEY
+UTILS_SECRET=$UTILS_SECRET
+DATABASE_URL=$DATABASE_URL
+REDIS_URL=redis://localhost:6379
+URL=$URL
+FILE_STORAGE=local
+FILE_STORAGE_LOCAL_ROOT_DIR=$FILE_STORAGE_LOCAL_ROOT_DIR
```

Comment:
/var/lib/outline/data instead of $FILE_STORAGE_LOCAL_ROOT_DIR

Or another question, I like it when everything is in opt, is it better to have an /opt/outline_data folder?

Reply:
Hmm I had a look at the Filesystem Hierarchy Standard, and it says `/var/opt/outline` should be used (https://www.pathname.com/fhs/pub/fhs-2.3.html#VAROPTVARIABLEDATAFOROPT), but many people use `/var/lib` since that's what they're used to (https://www.pathname.com/fhs/pub/fhs-2.3.html#VARLIBVARIABLESTATEINFORMATION)